### DISCUSSION TITLE
Use Ender 6 Display directly with a Raspberry Pi

### DISCUSSION THREAD

**Discussion by xsnoopy** [Desuuuu/klipper#130] [Ideas]
Hi, 

I have the Ender 6 with the Creality V4.3.1 board. I was thinking to alter your code in order to make the Display working directly connected to a Raspberry Pi. 

This would give some advantages:
1. You could stay with the standard Klipper Firmware.
2. You would have the 10pin connector available for other things ( at least 2 GPIOs are on that pin header)
3. You could change the Creatlity board and stay with the Stock Display. 

Problems: The display seems to have a to low resolution to run klipper screen. 

My idea is the use the moonraker API and write a connection between the Raspbery Pi and the Ender 6 display.

If I see that correct your screens are pretty stationary, you just load new screens and replace some text variables. 

It would be great if you give me your thoughts and could help me with a jump start to understand how the display is reading them.

**Comment by terrafirma2021** [comment]
Hi mate its already been done with a T5L via USD TTL
https://www.youtube.com/watch?v=_WQkVF006PY&t=904s
